tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/Pcm2JavaTransformationTest.xtend

```diff
@@ -0,0 +1,194 @@
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.JavaQueryUtil.*
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.TransformationDirectionConfiguration.configureUnidirectionalExecution
+import static tools.vitruv.testutils.matchers.ModelMatchers.*
+
+class Pcm2JavaTransformationTest extends ViewBasedVitruvApplicationTest {
+	protected var extension Pcm2JavaViewFactory viewFactory
+	
+	// === setup ===
+	
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_MODEL_FILE_EXTENSION = "pcm"
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_SYSTEM_FILE_EXTENSION = "system"
+	@Accessors(PROTECTED_GETTER)
+	static val PCM_MODEL_NAME = "model"
+	@Accessors(PROTECTED_GETTER)
+	static val JAVA_MODEL_FILE_EXTENSION = "java"
+	@Accessors(PROTECTED_GETTER)
+	static val JAVA_MODEL_NAME = "code"
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_FOLDER_NAME = "model"
+	
+	@BeforeEach
+	def final void setupViewFactory(){
+		viewFactory = new Pcm2JavaViewFactory(virtualModel);
+	}
+	
+	@BeforeEach
+	def setupTransformatonDirection() {
+		configureUnidirectionalExecution(virtualModel)
+	}
+	
+	@BeforeEach
+	def setupJavaClasspath() {
+		JavaSetup.resetClasspathAndRegisterStandardLibrary();
+	}
+	
+	@BeforeEach
+	def disableTransitiveChangePropagation() {
+		virtualModel.changePropagationMode = ChangePropagationMode.SINGLE_STEP
+	}
+
+	protected def Path getProjectModelPath(String modelName, String modelFileExtension) {
+		Path.of(MODEL_FOLDER_NAME).resolve(modelName + "." + modelFileExtension)
+	}
+
+	override protected getChangePropagationSpecifications() {
+		return #[new Pcm2JavaChangePropagationSpecification()]
+	}
+
+	protected def void createAndRegisterRoot(View view, EObject rootObject, URI persistenceUri) {
+		view.registerRoot(rootObject, persistenceUri)
+	}
+	
+	protected def void deleteAndUnregisterRoot(View view, EObject rootObject) {
+		EcoreUtil.delete(rootObject)
+	}
+	
+	// === creators ===
+	
+	private def void createRepository((Repository)=> void repositoryInitalization){
+		changePcmView[
+			var repository = RepositoryFactory.eINSTANCE.createRepository();
+			createAndRegisterRoot(repository, getProjectModelPath(PCM_MODEL_NAME, PCM_MODEL_FILE_EXTENSION).uri)
+			repositoryInitalization.apply(repository)
+		]
+	}
+	
+	protected def void createRepostory(String repositoryName){
+		createRepository[
+			entityName = repositoryName;
+		]
+	}
+	
+	private def void createSystem((System) => void systemInitialization){
+		changePcmView[
+			val system = SystemFactory.eINSTANCE.createSystem();
+			createAndRegisterRoot(system, getProjectModelPath(PCM_MODEL_NAME, PCM_SYSTEM_FILE_EXTENSION).uri)
+			systemInitialization.apply(system)
+		]
+	}
+	
+	protected def void createSystem(String systemyName){
+		createSystem [
+			entityName = systemyName;
+		]
+	}
+	
+	// === assertions ===
+	
+	protected def void assertCompilationUnits(View view, Collection<CompilationUnit> expectedCompilationUnits){
```

Comment:
From the name and common JUnit assert methods, I expected the method to compare the passed compilation units for equality. To better convey its functionality, maybe rename it to something like `assertExistenceOfCompilationUnitsDeeplyEqualTo` or similar?